nkansal96/aurora-kitchen-helper
# Connect the Python program, Express server, and React Frontend

# This is a big PR

It does a few things:

- cleans up a bunch of code we weren't using
- gets everything passing the linter
- adds a couple relevant tests (should add more later though)
- updates `server.py` to be more compliant with python standards
- changes the visual styling of the frontend
- *most importantly*: connects the python server to the client app

The most important changes are in the following files:

- [HomePage.js](https://github.com/nkansal96/aurora-kitchen-helper/pull/4/files#diff-416686125dada6819460e15f5a3a079c)
- [client/server/index.js](https://github.com/nkansal96/aurora-kitchen-helper/pull/4/files#diff-3678dbf1c4cb39376069ea70e28d54f5)
- [server/server.py](https://github.com/nkansal96/aurora-kitchen-helper/pull/4/files#diff-762988684ae9cb15fd2b07a83e11a4e1)

## How it works

Basically, instead of writing the data from Aurora to a file,
the Python server now sends the Aurora data over to the express server through a POST request to the `/updateData` route.
The frontend then fetches this data periodically via a GET request to the `/data` route.

This eliminates the need to store the data on disk in a file and simplifies the communication between the different components.

## DON'T MERGE YET

**The Aurora Python stuff doesn't seem to work on Linux at the moment, so I wasn't able to test the integration between the Python program and the Express server. Someone should confirm that this is working before this can be merged.**

### How to test this?

1. check out this branch
2. install the dependencies
    - `cd client`
    - `npm install`
3. run the server
    - `npm start`
4. run the python program
    - `cd server` (from the repo root)
    - `python server.py`
5. open the frontend at [localhost:3000](http://localhost:3000)
6. talk to your computer
7. observe errors or changes on the web app frontend

## Screenshots

#### Before
![screenshot_20180606_151059](https://user-images.githubusercontent.com/2906913/41067840-e344a1b0-699b-11e8-86b8-e8e2c33a7e5e.png)

#### After
![screenshot_20180606_145645](https://user-images.githubusercontent.com/2906913/41067839-e327cdba-699b-11e8-83ff-0f74701e24cf.png)
